packages/expo-router/src/toolbar/elements.tsx

```diff
@@ -385,6 +387,81 @@ export const ToolbarSpacer = (props: ToolbarSpacerProps) => {
   );
 };
 
+export interface ToolbarSearchBarProps extends SearchBarProps {
+  /**
+   * Whether to hide the shared background when `sharesBackground` is enabled.
+   *
+   * @see [Official Apple documentation](https://developer.apple.com/documentation/uikit/uibarbuttonitem/hidessharedbackground) for more information.
+   *
+   * @platform iOS 26+
+   */
+  hidesSharedBackground?: boolean;
+  /**
+   * Whether the search bar should be hidden.
+   *
+   * @default false
+   */
+  hidden?: boolean;
+  /**
+   * Whether the search bar shares the background with adjacent toolbar items.
+   *
+   * @see [Official Apple documentation](https://developer.apple.com/documentation/uikit/uibarbuttonitem/sharesbackground) for more information.
+   *
+   * @platform iOS 26+
+   * @default false
+   */
+  sharesBackground?: boolean;
+}
+
+/**
+ * A search bar component for the toolbar.
+ * It should only be used as a child of `Toolbar`.
+ *
+ * > **Note**: The stack header must be visible in order to add a search bar.
+ * > Once `Toolbar.SearchBar` is used, the search bar will no longer be available
+ * > in this screen’s header.
+ *
+ * @example
+ * ```tsx
+ * <Toolbar>
+ *   <Toolbar.SearchBar />
+ *   <Toolbar.Spacer />
+ *   <Toolbar.Button icon="mic" />
+ * </Toolbar>
+ * ```
+ *
+ * @see [React Native Screens SearchBarProps](https://github.com/software-mansion/react-native-screens/blob/main/src/types.tsx#L822) for more information about available props.
+ *
+ * @platform ios 26+
```

Comment:
So on older versions you need to use stack searchbar but it can have placement that puts it in the footer location? Maybe these should be unified somehow...